rust/main/agents/scraper/src/agent.rs

```diff
@@ -137,8 +139,11 @@ impl BaseAgent for Scraper {
 impl Scraper {
     /// Sync contract data and other blockchain with the current chain state.
     /// This will spawn long-running contract sync tasks
-    async fn scrape(&self, domain_id: u32) -> Instrumented<JoinHandle<()>> {
-        let scraper = self.scrapers.get(&domain_id).unwrap();
+    async fn scrape(&self, domain_id: u32) -> eyre::Result<Instrumented<JoinHandle<()>>> {
+        let scraper = self.scrapers.get(&domain_id).ok_or_else(|| {
+            tracing::error!(domain_id, "Failed to get scraper for domain id");
+            eyre::eyre!("Failed to get scraper for domain id")
+        })?;
```

Comment:
This method is invoked from iteration over `scrapers` hashmap. I wonder if it makes sense to pass `ChainScraper` as an argument for this method instead of re-retrieving it from the map and handling the error?

Reply:
Yeah good idea :+1: